### PR TITLE
Add N-gram support in `FullText::Contains`

### PR DESCRIPTION
**Pull request by ghost**
### Add N-gram support in `FullText::Contains` 

Now users can use FullText::Contains with ngram and edge_ngram indexes. That enables wildcard queries like `"Quan*ati*"` which will match for example `Quantization`

### Changelog category 

* New feature

### Description for reviewers 

...

### CODE REVIEW

File: ydb/core/kqp/runtime/kqp_full_text_source.cpp [hunk 1]
```diff
@@ -946,6 +971,36 @@ class TFullTextContainsSource : public TActorBootstrapped<TFullTextContainsSourc
...
+                std::back_inserter(ucs4Pattern));
+
+            auto regex = NPire::TLexer(ucs4Pattern.begin(), ucs4Pattern.end())
+                .SetEncoding(NPire::NEncodings::Utf8())
+                .Parse().Compile<NPire::TScanner>();
+
+            PostfilterMatchers.push_back([regex=std::move(regex)](const TStringBuf str) {
+                return Pire::Matches(regex, str);
+            });
+        }
```
[the remark this answers is not in the document]
Comment (ghost) [reply ydb/core/kqp/runtime/kqp_full_text_source.cpp:997]: I think it's okay

File: ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp [hunk 2]
```diff
@@ -2598,6 +2581,169 @@ Y_UNIT_TEST(SelectWithFulltextContainsAndSnowball) {
...
 }
 
+Y_UNIT_TEST_TWIN(SelectWithFulltextContainsAndNgram, Edge) {
+    auto kikimr = Kikimr();
+    auto db = kikimr.GetQueryClient();
+
+    NYdb::NQuery::TExecuteQuerySettings querySettings;
+    querySettings.ClientTimeout(TDuration::Minutes(1));
+
+    CreateTexts(db);
```
Comment (azevaykin) [inline ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp]: There should be a test with Utf8 column

File: ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp [hunk 3]
```diff
@@ -2598,6 +2581,169 @@ Y_UNIT_TEST(SelectWithFulltextContainsAndSnowball) {
...
+
+    {
+        TString query = R"sql(
+            UPSERT INTO `/Root/Texts` (`Key`, `Text`) VALUES
+                (0, "Arena Allocation"),
+                (1, "Area Renaissance"),
+                (2, "Werner Heisenberg"),
+                (3, "Bern city"),
+                (4, "lusedaedae"),
+                (5, "lusedaeda")
```
Comment (azevaykin) [inline ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp]: Let's add 2 rows: empty string and NULL string

File: ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp [hunk 4]
```diff
@@ -2598,6 +2581,169 @@ Y_UNIT_TEST(SelectWithFulltextContainsAndSnowball) {
     }
 }
 
+Y_UNIT_TEST_TWIN(SelectWithFulltextContainsAndNgram, Edge) {
```
Comment (azevaykin) [inline ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp]: Lets add more tests:
1. Test with Utf8 column and multi-byte UTF-8 characters like emoji, Chinese characters. Verify that n-gram sizes are based on character count, not byte count
2. Test for regex special characters in wildcard patterns.
3. The current tests only use 3-3 ngrams. Missing tests for: 2-2 ngrams, 4-4 ngrams, variable length ngrams (e.g., 2-4, 3-5)
4. Boundary Cases: words shorter than ngram minimum length, words exactly at ngram boundaries.
5. Index with covered columns.
6. Table contains empty and NULL strings.
7. Specific tests for edge ngrams.

**Review by azevaykin — CHANGES_REQUESTED** [review]

File: ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp (hunk 4 above)
Comment (ghost) [reply ydb/core/kqp/ut/indexes/kqp_indexes_fulltext_ut.cpp]: Done!

**Review by azevaykin — APPROVED** [review]

**Review by gridnevvvit — APPROVED** [review]
